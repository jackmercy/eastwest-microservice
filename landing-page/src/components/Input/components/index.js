export { default as FieldLabel } from './FieldLabel';
export { default as FieldError } from './FieldError';
export { default as FieldInput } from './FieldInput';
