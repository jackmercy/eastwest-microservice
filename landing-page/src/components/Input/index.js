export * from './components';

export { default } from './Input';
export { default as InputNumber } from './InputNumber';
export { default as InputPhone } from './InputPhone';
export { default as InputSearch } from './InputSearch';
export { default as InputPassword } from './InputPassword';
