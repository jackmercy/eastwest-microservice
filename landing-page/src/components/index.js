export { default as SEO } from './SEO';
export { default as Logo } from './Logo';
export { default as Modal } from './Modal';
export { default as Input } from './Input';
export { default as Select } from './Select';
export { default as Button } from './Button';
export { default as Loading } from './Loading';
export { default as Carousel } from './Carousel';
export { default as Textarea } from './Textarea';
export { default as CheckBox } from './CheckBox';
export { default as PopupMenu } from './PopupMenu';
export { default as StaticHTML } from './StaticHTML';

export * from './Input';
export * from './Modal';
