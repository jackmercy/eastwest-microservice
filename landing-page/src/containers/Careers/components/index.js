export { default as SectionTitle } from './SectionTitle';
export { default as MenuScrollableLayout } from './MenuScrollableLayout';
