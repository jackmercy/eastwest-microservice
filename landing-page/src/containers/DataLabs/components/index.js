export { default as Teaser } from './Teaser';
export { default as JoinUs } from './JoinUs';
export { default as Become } from './Become';
export { default as HowItWorks } from './HowItWorks';
export { default as ReadyToStart } from './ReadyToStart';
