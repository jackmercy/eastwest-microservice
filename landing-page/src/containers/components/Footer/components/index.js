export { default as Register } from './Register';
export { default as CopyRight } from './CopyRight';
export { default as SummaryInfo } from './SummaryInfo';
export { default as SectionTitle } from './SectionTitle';
export { default as SwitchLanguages } from './SwitchLanguages';
