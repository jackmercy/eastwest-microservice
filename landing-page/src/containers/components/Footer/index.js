import dynamic from 'next/dynamic';

export default dynamic(() => import('./Footer'), { ssr: false });
