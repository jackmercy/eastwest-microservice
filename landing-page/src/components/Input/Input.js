import React, { memo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { FieldLabel, FieldInput, FieldError } from './components';
import useStyles from './input.styles';

const Input = ({
  name,
  error,
  label,
  dataTest,
  required,
  className,
  inputClassName,
  labelClassName,
  errorClassName,
  ...restProps
}) => {
  const classes = useStyles();

  return (
    <div
      className={classNames(classes.root, className)}
      data-test={dataTest || 'component-input-root'}
    >
      {label ? (
        <FieldLabel
          label={label}
          htmlFor={name}
          required={required}
          className={labelClassName}
        />
      ) : null}
      <FieldInput
        name={name}
        hasError={!!error}
        required={required}
        className={inputClassName}
        {...restProps}
      />
      {error ? <FieldError error={error} className={errorClassName} /> : null}
    </div>
  );
};

Input.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.node,
  error: PropTypes.string,
  required: PropTypes.bool,
  dataTest: PropTypes.string,
  className: PropTypes.string,
  inputClassName: PropTypes.string,
  labelClassName: PropTypes.string,
  errorClassName: PropTypes.string,
};
Input.defaultProps = {
  error: '',
  dataTest: '',
  className: '',
  required: false,
  label: undefined,
  inputClassName: '',
  labelClassName: '',
  errorClassName: '',
};

export default memo(Input);
