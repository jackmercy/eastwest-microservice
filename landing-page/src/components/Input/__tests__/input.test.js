import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { FieldLabel, FieldInput, FieldError } from '../components';
import Input from '../Input';

describe('Testing Input component', () => {
  let wrapper;
  const props = { label: 'input', name: 'input', error: 'some error' };

  beforeAll(() => {
    wrapper = mount(<Input {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<Input {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have a FieldLabel', () => {
    expect(wrapper.find(FieldLabel)).toHaveLength(1);
  });

  it('should have a FieldInput', () => {
    expect(wrapper.find(FieldInput)).toHaveLength(1);
  });

  it('should have a FieldLabel', () => {
    expect(wrapper.find(FieldError)).toHaveLength(1);
  });
});
