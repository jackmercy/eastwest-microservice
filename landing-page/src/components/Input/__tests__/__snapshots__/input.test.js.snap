// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component should render match snapshot 1`] = `
<div
  className="root-0-2-1"
  data-test="component-input-root"
>
  <Memo(FieldLabel)
    className=""
    htmlFor="input"
    label="input"
    required={false}
  />
  <Memo(FieldInput)
    className=""
    hasError={true}
    name="input"
    required={false}
  />
  <Memo(FieldError)
    className=""
    error="some error"
  />
</div>
`;
